### ISSUE TITLE
Shooting switches not working at all

### ISSUE THREAD

**Issue by bfg01** [atsb/Doom64EX-Plus#36]
This is basically issue #2 , but I just think it's an issue *bigger* than we thought...

Shooting switches is essential part of gameplay ever since classic Doom. Let alone the "secrets" stuff, IIRC there are some levels which require this feature to even be clearable (I spent good part of the weekend testing playing through several levels).
And no, the noclip cheat wouldn't necessarily help since not all of them trigger doors, neither can they be activated by pressing the "use" key.
Most clear example that comes to mind now: MAP29.

I already read that you mentioned that you've reviewed the source code several times already, but, couldn't we try giving this another try? I think this would be kind of higher priority...
It was possible to fix the **ugly** segmentation fault thing when building on "stable" distros, and also there are more contributors. Perhaps there could be better luck this time?

Oh, and small additional side note: it seems that Doom 64 introduced item picking as "secret trigger" as well, and not just the "sectors". However, IIRC, I noticed that, after clearing a map, seemingly only the secret "sectors" are counted...

Thanks again.

**Comment by atsb** [comment]
Thanks for the testing, now this is interesting.  So it isn't just related to shooting linedefs but all switches.  At least this helps narrow it down as I was looking purely at the shootable linedef code.  I was testing mostly for 32bit Windows as that is what I provide as a workaround to the issue.

I'll take a look again

**Comment by atsb** [comment]
Ok, I found the bad commit from 14 days ago.  It was me, changing the long integers to standard int32 and int's.  Turns out, it actually needs the long's there to do some calculation based on the bit-width of the system used.  So having it all as 32-bit wide on all systems actually broke it for 64bit systems....

Pulling from master will give you some joy now :)

Edit: I hardened this code with intptr_t as it is wide enough for all systems and all bit-widths.  We're golden, everything should work fine now.
